### PR TITLE
Set up Travis CI

### PR DESCRIPTION
**Pull request by podcastinator**
A minimal Travis CI setup that tests the CMake build.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+dist: bionic
+
+env:
+  matrix:
+    # cmake test
+    - TASK=cmake_test
+
+before_install:
+  # Configure Intel and Microsoft APT repositories
+  - sudo apt-get -q update
```
Comment (chester-leung) [inline .travis.yml:13]: Travis CI's [docs](https://docs.travis-ci.com/user/installing-dependencies) suggest not to run `apt-get upgrade` as it'll increase build times significantly:

"Do not run apt-get upgrade in your build as it downloads up to 500MB of packages and significantly extends your build time. Additionally, some packages may fail to update, which will lead to a failed build."

Is there any way around doing this for us?

**Comment by chester-leung** [comment]
Very exciting! Left a comment.

File: .travis.yml (hunk 1 above)
Comment (podcastinator) [reply .travis.yml:13]: Oh we do an update, not an upgrade.

Comment (chester-leung) [reply .travis.yml:13]: Oops. My bad.

**Review by chester-leung — APPROVED** [review]

**Pull request merged**